testing/test_recwarn.py

```diff
@@ -284,3 +284,23 @@ def test(run):
         ''')
         result = testdir.runpytest()
         result.stdout.fnmatch_lines(['*2 passed in*'])
+
+    def test_match_regex(self):
+        with pytest.warns(UserWarning, match=r'must be \d+$'):
+            warnings.warn("value must be 42", UserWarning)
+
+        with pytest.raises(pytest.fail.Exception):
```

Comment:
Please include a case which matches with `match` but the warning raised is wrong:

```python
with pytest.raises(pytest.fail.Exception):
   with pytest.warns(UserWarning, match=r'must be \d+$'):
       warnings.warn("value must be 42", RuntimeWarning)
```

Reply:
done.